client/functions.lua

```diff
@@ -109,31 +109,38 @@ local function isVehicleInRange(vehicle, maxDistance)
     end
 end
 
----Will be execuded when the opening of the lock succeeds.
----@param vehicle number The entity number of the vehicle.
----@param plate string The plate number of the vehicle.
-local function lockpickSuccessCallback(vehicle, plate)
-    TriggerServerEvent('hud:server:GainStress', math.random(1, 4))
-
-    if cache.seat == -1 then
-        TriggerServerEvent('qb-vehiclekeys:server:AcquireVehicleKeys', plate)
+---Chance to destroy lockpick
+---@param isAdvancedLockedpick any
+---@param vehicleClass any
+local function breakLockpick(isAdvancedLockedpick, vehicleClass)
+    local chance = math.random()
+    if isAdvancedLockedpick then -- there is no benefit to using an advanced tool in the default configuration.
+        if chance <= config.removeAdvancedLockpickChance[vehicleClass] then
+            TriggerServerEvent("qb-vehiclekeys:server:breakLockpick", "advancedlockpick")
+        end
     else
-        exports.qbx_core:Notify(locale("notify.vehicle_lockedpick"), 'success')
-        TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(vehicle), 1)
-        Entity(vehicle).state.isOpen = true
+        if chance <= config.removeNormalLockpickChance[vehicleClass] then
+            TriggerServerEvent("qb-vehiclekeys:server:breakLockpick", "lockpick")
+        end
     end
 end
 
+---Will be executed when the lock opening is successful.
+---@param vehicle number The entity number of the vehicle.
+local function lockpickSuccessCallback(vehicle)
+    TriggerServerEvent('hud:server:GainStress', math.random(1, 4))
+    exports.qbx_core:Notify(locale("notify.vehicle_lockedpick"), 'success')
+    TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(vehicle), 1)
+    Entity(vehicle).state.isOpen = true
```

Comment:
Did this state bag already exist? Isn't the vehicle lock state already synced via natives?

Reply:
This flag is to prevent the vehicle from being locked and it has been before